substrate-archive/src/archive.rs

```diff
@@ -53,16 +53,18 @@ pub struct ChainConfig {
 	pub(crate) rocksdb_secondary_path: Option<PathBuf>,
 	/// Chain spec.
 	#[serde(skip)]
-	pub(crate) spec: Option<ChainSpecConfig>,
+	pub(crate) spec: Option<Box<dyn ChainSpec>>,
```

Comment:
Is the `spec` still optional after this?

Reply:
I think it should remain optional in the archive lib, and then each individual library can enforce it if it needs to. Polkadot Archive must enforce that chain_spec exists because it uses `code_substitute` directly for instance, but another chain that does not rely on it won't have to.

Now that I consider it, it might make sense to just require chain spec regardless though